blocks/main_loop.py

```diff
@@ -156,16 +159,17 @@ def _check_finish_training(self):
         # the iteration the corresponding log record can be found only in
         # the previous row.
         if (self.log.current_row.training_finish_requested or
-                self.log.current_row.keyboard_interrupt_received or
-                self.log.previous_row.keyboard_interrupt_received):
+                self.log.current_row.interrupt_received or
+                self.log.previous_row.interrupt_received):
             raise TrainingFinish
 
-    def _handle_keyboard_interrupt(self, signal_number, frame):
+    def _handle_interrupt(self, signal_number, frame):
         # After receiving a first keyboard interrupt signal,
         # ignore all following ones.
```

Comment:
Yeah, but that's the current behavior, right? We restore the handler to the original one in the next few lines, and the original handler makes Python quite immediately, so pressing Ctrl + C twice should make things crash with a `KeyboardInterrupt` straight away (which I think is the correct behavior). I've done this a few times (just press Ctrl + C a few times, sometimes it doesn't catch the second one) and it works just fine.